apps/emqx_management/src/emqx_mgmt_api_clients.erl

```diff
@@ -1224,10 +1224,20 @@ subscribe(#{clientid := ClientID, topic := Topic} = Sub) ->
     end.
 
 subscribe_batch(#{clientid := ClientID, topics := Topics}) ->
-    %% We use emqx_channel instead of emqx_channel_info (used by the emqx_mgmt:lookup_client/2),
-    %% as the emqx_channel_info table will only be populated after the hook `client.connected`
-    %% has returned. So if one want to subscribe topics in this hook, it will fail.
-    case ets:lookup(?CHAN_TAB, ClientID) of
+    %% On the one hand, we first try to use `emqx_channel' instead of `emqx_channel_info'
+    %% (used by the `emqx_mgmt:lookup_client/2'), as the `emqx_channel_info' table will
```

Comment:
emqx_mgmt:lookup_client polls all nodes, should maybe check channel registry to find the pid and node(Pid) to find the node ?

Reply:
that is, when channel registry is enabled.
if not, it's still necessary to loop-over all running nodes

Reply:
It seems we still support `broker.enable_session_registry = false`, in which case we can't rely on the global registry?

Reply:
Ah, when I replied, the 2nd message wasn't visible. Sounds good.